pandas/core/groupby/generic.py

```diff
@@ -1418,17 +1418,16 @@ def transform(self, func, *args, **kwargs):
         # and deal with possible broadcasting below.
         result = getattr(self, func)(*args, **kwargs)
 
-        # a reduction transform
-        if not isinstance(result, DataFrame):
```

Comment:
The fix here seems oddly specific to only affect `fillna` - would things like `shift` and `cumsum` not be affected as well?

We already have a list of transforming / reducing operations in groupby.base so I think should be leveraging that here if required

Reply:
Thanks for the heads-up. In fact by adding the comparison of indices I meant exactly to determine if ``func`` is a reduction operation. Now that I know `groupby.base` has it, it is just perfect.